src/python/pants/backend/python/tasks/python_tool_prep_base.py

```diff
@@ -130,7 +131,12 @@ def execute(self):
     tool_subsystem = self.tool_subsystem_cls.scoped_instance(self)
 
     interpreter_cache = PythonInterpreterCache.global_instance()
-    interpreter = min(interpreter_cache.setup(filters=tool_subsystem.get_interpreter_constraints()))
+    interpreters = interpreter_cache.setup(filters=tool_subsystem.get_interpreter_constraints())
+    if not interpreters:
+      raise TaskError('Found no Python interpreter capable of running the {} tool with '
```

Comment:
(Feel free to ignore this suggestion)

```suggestion
      raise TaskError('Could not find a Python interpreter capable of running the {} tool with '
```